packages/manager/modules/nasha/src/dashboard/partition/snapshots/snapshots.routing.js

```diff
@@ -1,45 +1,40 @@
 import { createTaskTrackerStateOptions } from '../../../components/task-tracker';
-import { STATE_NAME } from './snapshots.constants';
+import { STATE_NAME, TASK_TRACKER_STATE_NAME } from './snapshots.constants';
 
 export default /* @ngInject */ ($stateProvider) => {
-  const snapshotEnumKey = 'dedicated.storage.SnapshotEnum';
-
   $stateProvider.state(STATE_NAME, {
     url: '/snapshots',
     component: 'nashaDashboardPartitionSnapshots',
     resolve: {
       breadcrumb: () => null,
+      close: /* @ngInject */ (goBack, trackTasks) => ({
+        tasks,
+        partitionName,
+        customSnapshotName,
+        error,
+      } = {}) =>
+        tasks
+          ? trackTasks({ tasks, partitionName, customSnapshotName })
+          : goBack({ stateName: STATE_NAME, error }),
       customSnapshots: /* @ngInject */ ($http, partitionApiUrl) =>
         $http.get(`${partitionApiUrl}/customSnapshot`).then(({ data }) => data),
       goToDelete: /* @ngInject */ ($state, serviceName, partitionName) => (
         customSnapshotName,
       ) =>
-        $state.go('nasha.dashboard.partition.snapshots.delete', {
+        $state.go(`${STATE_NAME}.delete`, {
           serviceName,
           partitionName,
           customSnapshotName,
         }),
-      snapshots: /* @ngInject */ (
-        $http,
-        partitionApiUrl,
-        prepareSnapshots,
-        customSnapshots,
-        snapshotEnum,
-      ) =>
-        $http
-          .get(`${partitionApiUrl}/snapshot`)
-          .then(({ data: snapshots }) =>
-            prepareSnapshots(snapshots, customSnapshots, snapshotEnum),
-          ),
-      snapshotEnum: /* @ngInject */ (schema) =>
-        schema.models[snapshotEnumKey].enum,
+      snapshotTypes: /* @ngInject */ ($http, partitionApiUrl) =>
+        $http.get(`${partitionApiUrl}/snapshot`).then(({ data }) => data),
       trackTasks: /* @ngInject */ ($state) => (params) =>
-        $state.go(`${STATE_NAME}.task-tracker`, params),
+        $state.go(TASK_TRACKER_STATE_NAME, params),
```

Comment:
**nitpick**: function name suggests it does something related to tracking while it does a state navigation. Can you rename according to its responsibility?